src/main/java/org/assertj/core/internal/Maps.java

```diff
@@ -637,8 +661,10 @@ public void assertHasSameSizeAs(AssertionInfo info, Map<?, ?> map, Map<?, ?> oth
    * @throws AssertionError if the given {@code Map} does not contain the given keys or if the given {@code Map}
    *           contains keys that are not in the given array.
    */
+  // varargs are not used in order to avoid a "Possible heap pollution" warning.
+  // The method cannot be made final and annotated with @SafeVarargs because the class is mocked in tests
```

Comment:
I'm currently doing some cleanup in this class and I would like to remove these comments as the callers don't have vararg parameters (e.g., `ForProxy` methods).

Nothing suggests that we should switch to varargs in internal classes so I don't see much value in keeping them. If you don't mind, I'll remove them.

Reply:
I agree, this comment is only relevant in the API layer.

Reply:
I am completely fine in removing the comments. I put them there for future reviews of the class in case someone wonders why `varargs` or not used